### ISSUE TITLE
Empty root tag into `List` deserialization bug

### ISSUE THREAD

**Issue by Seatec13** [FasterXML/jackson-dataformat-xml#319]
/*copied from https://github.com/FasterXML/jackson-databind/issues/2163 */

Hello,
I have a strange problem with jackson-dataformat 2.9.* (with 2.8.* everything work good):

Data class: (Lombok -- modify for test)

```
@Data
public class PackInstanceOrder {
    private Long orderId;
    private Long orderTypeId;
}
```

Simple test (Kotlin?):

```
    @Test
    fun test() {
        val XML_OBJECT_MAPPER = XmlMapper()

        val xml = "<orders></orders>"Labels
        val orders: List<PackInstanceOrder> = XML_OBJECT_MAPPER.readValue(xml,
            XML_OBJECT_MAPPER.getTypeFactory().constructCollectionType(List::class.java, PackInstanceOrder::class.java))

        assertThat(orders, notNullValue())
    }
```

throw an error:

```
    com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "" (class PackInstanceOrder), not marked as ignorable (2 known properties: "orderId", "orderTypeId"])
    at [Source: (StringReader); line: 1, column: 18] (through reference chain: java.util.ArrayList[0]->PackInstanceOrder[""])
```

When adding @JsonIgnoreProperties(ignoreUnknown = true) annotation to PackInstanceOrder, error was changed:

```
    com.fasterxml.jackson.databind.exc.MismatchedInputException: Unexpected end-of-input when binding data into PackInstanceOrder
    at [Source: (StringReader); line: 1, column: 18] (through reference chain: java.util.ArrayList[1])
```

BUT! When I change xml to

    <orders/>

test didn't throw any errors and pass well.

**Comment by cowtowncoder** [comment]
Yes, can reproduce this; adding a (failing) unit test.

**Comment by cowtowncoder** [comment]
For some reason, seems to work for `master`. Possibly fixed. But that's just for 3.0 (long ways still); problem reproducible with 2.10.

**Comment by Reamer** [comment]
Is there any plan to back port a potential fix?

**Comment by cowtowncoder** [comment]
@Reamer if someone can figure out how, sure. Likely would go in 2.11, unless it's a trivial change (in which could go in 2.10.x patch). I don't think active work is going on for this so help would be appreciated.

**Comment by Reamer** [comment]
Hi @cowtowncoder,
Sorry, I have no deep knowledge in jackson and it's components to provide any help. I can provide [dependency-check-sonar-plugin](https://github.com/dependency-check/dependency-check-sonar-plugin) as an example project, maybe that's helpful.

**Comment by cowtowncoder** [comment]
@Reamer no problem, and thank you for the offer: we do have a failing (in 2.x) test already, so that's fine. Only need to have time to dig in, hope to find solution for 2.x (and maybe understand what changes to 3.0 are relevant).

**Comment by cowtowncoder** [comment]
Have not been able to fix, but I have bit better idea on what is going on, and hope to fix this for 2.11. Challenge is that of coercing empty XML String into "empty" value of POJO, Map or List, as a special case, being accepted as empty/null when not being bound to String (or scalar)
